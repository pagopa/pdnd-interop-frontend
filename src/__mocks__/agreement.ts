import { AgreementSummary } from '../../types'

export const agreementActiveEservicePublished: AgreementSummary = {
  id: 'dsfjds-jojoi-jdsfds',
  state: 'ACTIVE',
  eservice: {
    name: 'Riscossione TARI',
    id: 'osdijf-dsjfdisj-jsdfdsj',
    descriptorId: 'lskdfok-jisjdfs-djdsjfn',
    version: '4',
    state: 'PUBLISHED',
  },
  eserviceDescriptorId: 'dnsoifn-dsfjdsiof-dsjfsd',
  consumer: {
    name: 'Comune di Milano',
    id: 'afd_dss_cds',
  },
  producer: {
    name: 'Comune di Bologna',
    id: 'lds_kji_dsj',
  },
  attributes: [],
  suspendedByProducer: false,
  suspendedBySubscriber: false,
}

export const agreementSuspendedByProducerEservicePublished: AgreementSummary = {
  id: 'dsfjds-jojoi-jdsfds',
  state: 'SUSPENDED',
  eservice: {
    name: 'Riscossione TARI',
    id: 'osdijf-dsjfdisj-jsdfdsj',
    descriptorId: 'lskdfok-jisjdfs-djdsjfn',
    version: '4',
    state: 'PUBLISHED',
  },
  eserviceDescriptorId: 'dnsoifn-dsfjdsiof-dsjfsd',
  consumer: {
    name: 'Comune di Milano',
    id: 'afd_dss_cds',
  },
  producer: {
    name: 'Comune di Bologna',
    id: 'lds_kji_dsj',
  },
  attributes: [],
  suspendedByProducer: true,
  suspendedBySubscriber: false,
}

export const agreementSuspendedBySubscriberEservicePublished: AgreementSummary = {
  id: 'dsfjds-jojoi-jdsfds',
  state: 'SUSPENDED',
  eservice: {
    name: 'Riscossione TARI',
    id: 'osdijf-dsjfdisj-jsdfdsj',
    descriptorId: 'lskdfok-jisjdfs-djdsjfn',
    version: '4',
    state: 'PUBLISHED',
  },
  eserviceDescriptorId: 'dnsoifn-dsfjdsiof-dsjfsd',
  consumer: {
    name: 'Comune di Milano',
    id: 'afd_dss_cds',
  },
  producer: {
    name: 'Comune di Bologna',
    id: 'lds_kji_dsj',
  },
  attributes: [],
  suspendedByProducer: false,
  suspendedBySubscriber: true,
}
