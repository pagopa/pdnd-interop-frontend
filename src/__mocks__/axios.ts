import { AxiosStatic } from 'axios'

const axios = jest.genMockFromModule<AxiosStatic>('axios')

axios.create = jest.fn(() => axios)

export default axios
