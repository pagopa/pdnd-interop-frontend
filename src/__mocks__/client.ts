import { Client } from '../../types'

export const clientActiveAgreementActiveEservicePublished: Client = {
  id: 'dsfsdjf-sdfjsdfj-sdfjdsf',
  name: 'dipendenti del comune',
  description: 'il client dei dipendenti del mio comune',
  state: 'ACTIVE',
  agreement: {
    id: 'sjfaisds-sdfjsaodfj-sfajd',
    state: 'ACTIVE',
    descriptor: {
      id: 'djfiosj-dfjdsofj-dfjsdf',
      state: 'PUBLISHED',
      version: '5',
    },
  },
  eservice: {
    id: 'dsnodsi-sdfdsljf-sdjfodsjf',
    name: 'anagrafe comune di roma',
    provider: {
      institutionId: 'abc_fd_df',
      description: 'Comune di Roma',
    },
  },
  purposes: 'accesso pieno',
}

export const clientActiveAgreementActiveEserviceDeprecated: Client = {
  id: 'dsfsdjf-sdfjsdfj-sdfjdsf',
  name: 'dipendenti del comune',
  description: 'il client dei dipendenti del mio comune',
  state: 'ACTIVE',
  agreement: {
    id: 'sjfaisds-sdfjsaodfj-sfajd',
    state: 'ACTIVE',
    descriptor: {
      id: 'djfiosj-dfjdsofj-dfjsdf',
      state: 'DEPRECATED',
      version: '5',
    },
  },
  eservice: {
    id: 'dsnodsi-sdfdsljf-sdjfodsjf',
    name: 'anagrafe comune di roma',
    provider: {
      institutionId: 'abc_fd_df',
      description: 'Comune di Roma',
    },
    activeDescriptor: {
      id: 'sdjfsdjf-djfsdj-sdjfdsj',
      state: 'PUBLISHED',
      version: '7',
    },
  },
  purposes: 'accesso pieno',
}

export const clientSuspendedAgreementActiveEservicePublished: Client = {
  id: 'dsfsdjf-sdfjsdfj-sdfjdsf',
  name: 'dipendenti del comune',
  description: 'il client dei dipendenti del mio comune',
  state: 'SUSPENDED',
  agreement: {
    id: 'sjfaisds-sdfjsaodfj-sfajd',
    state: 'ACTIVE',
    descriptor: {
      id: 'djfiosj-dfjdsofj-dfjsdf',
      state: 'PUBLISHED',
      version: '5',
    },
  },
  eservice: {
    id: 'dsnodsi-sdfdsljf-sdjfodsjf',
    name: 'anagrafe comune di roma',
    provider: {
      institutionId: 'abc_fd_df',
      description: 'Comune di Roma',
    },
  },
  purposes: 'accesso pieno',
}

export const clientActiveAgreementSuspendedEservicePublished: Client = {
  id: 'dsfsdjf-sdfjsdfj-sdfjdsf',
  name: 'dipendenti del comune',
  description: 'il client dei dipendenti del mio comune',
  state: 'ACTIVE',
  agreement: {
    id: 'sjfaisds-sdfjsaodfj-sfajd',
    state: 'SUSPENDED',
    descriptor: {
      id: 'djfiosj-dfjdsofj-dfjsdf',
      state: 'PUBLISHED',
      version: '5',
    },
  },
  eservice: {
    id: 'dsnodsi-sdfdsljf-sdjfodsjf',
    name: 'anagrafe comune di roma',
    provider: {
      institutionId: 'abc_fd_df',
      description: 'Comune di Roma',
    },
  },
  purposes: 'accesso pieno',
}

export const clientActiveAgreementActiveEserviceSuspended: Client = {
  id: 'dsfsdjf-sdfjsdfj-sdfjdsf',
  name: 'dipendenti del comune',
  description: 'il client dei dipendenti del mio comune',
  state: 'ACTIVE',
  agreement: {
    id: 'sjfaisds-sdfjsaodfj-sfajd',
    state: 'ACTIVE',
    descriptor: {
      id: 'djfiosj-dfjdsofj-dfjsdf',
      state: 'SUSPENDED',
      version: '5',
    },
  },
  eservice: {
    id: 'dsnodsi-sdfdsljf-sdjfodsjf',
    name: 'anagrafe comune di roma',
    provider: {
      institutionId: 'abc_fd_df',
      description: 'Comune di Roma',
    },
  },
  purposes: 'accesso pieno',
}
