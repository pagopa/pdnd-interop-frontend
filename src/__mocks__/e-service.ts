import { EServiceReadType } from '../../types'

export const eservicePublished: EServiceReadType = {
  producer: {
    id: 'djofsi-sdfjdsi-djsfs',
    name: 'Comune di Bologna',
  },
  name: 'Nome e-service',
  description: 'Descrizione e-service',
  technology: 'REST',
  id: 'sdjof-sdfjdspof-dsfdsjf',
  state: 'PUBLISHED',
  descriptors: [],
  attributes: {
    verified: [],
    declared: [],
    certified: [
      {
        single: {
          id: 'dsdsld-dsdlds-lsdasdas',
          explicitAttributeVerification: false,
          verified: false,
          origin: 'dfkdsfk',
          code: 'dfjdso',
          name: 'Attributo 1',
          description: 'Descrizione attributo 1',
        },
      },
    ],
  },
}
