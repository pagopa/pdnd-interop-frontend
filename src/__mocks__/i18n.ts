import i18n from 'i18next'
import { initReactI18next } from 'react-i18next'
import HttpApi from 'i18next-http-backend'

i18n
  .use(initReactI18next)
  .use(HttpApi)
  .init({
    debug: false,
    fallbackLng: 'it',
    resources: {
      it: {
        common: {
          mode: {
            provider: 'erogazione',
            subscriber: 'fruizione',
          },
        },
        'shared-components': {
          pageReloadMessage: {
            message: 'Non siamo riusciti a recuperare i dati',
            cta: 'Ricarica la pagina',
          },
        },
        eservice: {
          create: {
            forwardWithSaveBtn: 'Salva bozza e prosegui',
            step1: {
              detailsTitle: 'Caratterizzazione E-Service',
              eserviceNameField: {
                label: "Nome dell'E-Service (richiesto)",
              },
              eserviceDescriptionField: {
                label: "Descrizione dell'E-Service (richiesto)",
              },
              attributes: {
                title: 'Attributi',
                addAttributeTable: {
                  noDataLabel: 'Nessun attributo presente',
                },
              },
            },
          },
        },
      },
    },
  })

export default i18n
