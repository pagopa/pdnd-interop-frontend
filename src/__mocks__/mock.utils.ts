import { Paginated } from '../api/comon.api.types'
import cloneDeep from 'lodash/cloneDeep'

/**
 * Wraps the data passed in the pagination data structure
 * */
export function createPaginatedMockData<T>(
  data: Array<T>,
  pagination?: Partial<Paginated<T>['pagination']>
): Paginated<T> {
  return {
    results: data,
    pagination: {
      offset: pagination?.offset ?? 0,
      limit: pagination?.offset ?? 50,
      totalCount: pagination?.offset ?? data.length,
    },
  }
}

/**
 * Create and returns a mock factory function
 */
export function createMockFactory<T>(defaultValue: T) {
  return (overwrites: Partial<T> = {}) => {
    return cloneDeep({
      ...defaultValue,
      ...overwrites,
    }) as T
  }
}
