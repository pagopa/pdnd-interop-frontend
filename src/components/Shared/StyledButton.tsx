import React, { FunctionComponent } from 'react'
import { Button } from '@mui/material'
import { StyledLink } from './StyledLink'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const StyledButtonComponent: FunctionComponent<any> = React.forwardRef(
  ({ children, isMock, ...props }, ref) => {
    return (
      <Button
        {...props}
        className={isMock ? 'mockFeature' : ''}
        component={props.to ? StyledLink : undefined}
        ref={ref}
      >
        {children}
      </Button>
    )
  }
)

StyledButtonComponent.displayName = 'StyledButton'

export const StyledButton = StyledButtonComponent
