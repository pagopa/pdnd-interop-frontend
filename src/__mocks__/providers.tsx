import React, { FunctionComponent, useState } from 'react'
import { Router } from 'react-router-dom'
import { DialogProps, LangCode, Party, ToastProps } from '../../types'
import {
  DialogContext,
  LangContext,
  LoaderContext,
  PartyContext,
  RoutesContext,
  TableActionMenuContext,
  ToastContext,
  TokenContext,
} from '../lib/context'
// import CssBaseline from '@mui/material/CssBaseline'
// import { ThemeProvider } from '@mui/material'
// import theme from '@pagopa/mui-italia/theme'
import { History, createMemoryHistory } from 'history'
import { getDecoratedRoutes } from '../lib/router-utils'
import { I18nextProvider } from 'react-i18next'
import i18n from './i18n'

type LangProviderProps = {
  defaultLang?: LangCode
}

type TokenProviderProps = {
  defaultToken?: string
}

type PartyProviderProps = {
  defaultParty?: Party
}

type TableActionMenuProviderProps = {
  defaultTableActionMenu?: string
}

type ToastProviderProps = {
  defaultToast?: ToastProps
}

type DialogProviderProps = {
  defaultDialog?: DialogProps
}

type LoaderProviderProps = {
  defaultLoader?: string
}

const LangProvider: FunctionComponent<LangProviderProps> = ({ children, defaultLang = 'it' }) => {
  const [lang, setLang] = useState<LangCode>(defaultLang)
  return <LangContext.Provider value={{ lang, setLang }}>{children}</LangContext.Provider>
}

const RoutesProvider: FunctionComponent = ({ children }) => {
  const allRoutes = getDecoratedRoutes()
  return <RoutesContext.Provider value={{ allRoutes }}>{children}</RoutesContext.Provider>
}

const TokenProvider: FunctionComponent<TokenProviderProps> = ({ children, defaultToken }) => {
  const [token, setToken] = useState<string | null>(defaultToken || null)
  return <TokenContext.Provider value={{ token, setToken }}>{children}</TokenContext.Provider>
}

const PartyProvider: FunctionComponent<PartyProviderProps> = ({ children, defaultParty }) => {
  const [party, setParty] = useState<Party | null>(defaultParty || null)
  const [availableParties, setAvailableParties] = useState<Array<Party> | null>(
    defaultParty ? [defaultParty] : null
  )
  return (
    <PartyContext.Provider value={{ party, setParty, availableParties, setAvailableParties }}>
      {children}
    </PartyContext.Provider>
  )
}

const TableActionMenuProvider: FunctionComponent<TableActionMenuProviderProps> = ({
  children,
  defaultTableActionMenu,
}) => {
  const [tableActionMenu, setTableActionMenu] = useState<string | null>(
    defaultTableActionMenu || null
  )
  return (
    <TableActionMenuContext.Provider value={{ tableActionMenu, setTableActionMenu }}>
      {children}
    </TableActionMenuContext.Provider>
  )
}

const ToastProvider: FunctionComponent<ToastProviderProps> = ({ children, defaultToast }) => {
  const [toast, setToast] = useState<ToastProps | null>(defaultToast || null)
  return <ToastContext.Provider value={{ toast, setToast }}>{children}</ToastContext.Provider>
}

const DialogProvider: FunctionComponent<DialogProviderProps> = ({ children, defaultDialog }) => {
  const [dialog, setDialog] = useState<DialogProps | null>(defaultDialog || null)
  return <DialogContext.Provider value={{ dialog, setDialog }}>{children}</DialogContext.Provider>
}

const LoaderProvider: FunctionComponent<LoaderProviderProps> = ({ children, defaultLoader }) => {
  const [loadingText, setLoadingText] = useState<string | null>(defaultLoader || null)
  return (
    <LoaderContext.Provider value={{ loadingText, setLoadingText }}>
      {children}
    </LoaderContext.Provider>
  )
}

type RouterProviderProps = {
  defaultHistory?: History<unknown>
}

export const AllTheProviders: FunctionComponent<
  LangProviderProps &
    TokenProviderProps &
    PartyProviderProps &
    TableActionMenuProviderProps &
    ToastProviderProps &
    DialogProviderProps &
    LoaderProviderProps &
    RouterProviderProps
> = ({
  children,
  defaultLang,
  defaultToken,
  defaultParty,
  defaultTableActionMenu,
  defaultToast,
  defaultDialog,
  defaultLoader,
  defaultHistory,
}) => {
  const history = defaultHistory || createMemoryHistory()

  return (
    <LangProvider defaultLang={defaultLang}>
      <I18nextProvider i18n={i18n}>
        <RoutesProvider>
          <TokenProvider defaultToken={defaultToken}>
            <PartyProvider defaultParty={defaultParty}>
              <Router history={history}>
                {/* <ThemeProvider theme={theme}> */}
                {/* <CssBaseline /> */}

                <TableActionMenuProvider defaultTableActionMenu={defaultTableActionMenu}>
                  <ToastProvider defaultToast={defaultToast}>
                    <DialogProvider defaultDialog={defaultDialog}>
                      <LoaderProvider defaultLoader={defaultLoader}>{children}</LoaderProvider>
                    </DialogProvider>
                  </ToastProvider>
                </TableActionMenuProvider>
                {/* </ThemeProvider> */}
              </Router>
            </PartyProvider>
          </TokenProvider>
        </RoutesProvider>
      </I18nextProvider>
    </LangProvider>
  )
}
