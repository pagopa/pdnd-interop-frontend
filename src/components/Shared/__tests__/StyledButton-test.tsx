import { noop } from 'lodash'
import React from 'react'
import renderer from 'react-test-renderer'
import { AllTheProviders } from '../../../__mocks__/providers'
import { Overlay } from '../Overlay'
import { StyledButton } from '../StyledButton'

describe('Snapshot', () => {
  it('matches button', () => {
    const component = renderer.create(<StyledButton onClick={noop}>Clicca qui</StyledButton>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches link', () => {
    const component = renderer.create(
      <AllTheProviders>
        <StyledButton to="/rotta-esempio">Clicca qui</StyledButton>
      </AllTheProviders>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
