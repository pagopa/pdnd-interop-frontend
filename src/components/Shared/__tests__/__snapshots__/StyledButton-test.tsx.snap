// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot matches button 1`] = `
<button
  className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Clicca qui
</button>
`;

exports[`Snapshot matches link 1`] = `
<a
  className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-606eve-MuiTypography-root-MuiLink-root-MuiButtonBase-root-MuiButton-root"
  href="/rotta-esempio"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
>
  Clicca qui
  <span
    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</a>
`;
